src/index.js

```diff
@@ -184,11 +190,16 @@ class WarmUP {
     return BbPromise.filter(allFunctions, (functionName) => {
       const functionObject = this.serverless.service.getFunction(functionName)
 
-      /** Function needs to be warm */
-      if (functionObject.warmup === true ||
+      const explicitlyIncluded = functionObject.warmup === true ||
         functionObject.warmup === this.options.stage ||
         (Array.isArray(functionObject.warmup) &&
-          functionObject.warmup.indexOf(this.options.stage) !== -1)) {
+          functionObject.warmup.indexOf(this.options.stage) !== -1)
+      const implicitlyIncluded = (this.warmup.includeAll === true ||
+        (Array.isArray(this.warmup.includeAll) &&
+          this.warmup.includeAll.indexOf(this.options.stage) !== -1)) &&
+        !(functionObject.warmup === false)
```

Comment:
Thinking a bit more...
What about the case where `includeAll` contains a stage and `functionObject` doesn't? Like include all contains staging & prod but the function only contains prod. What happens with staging? Currently it will get warmed. It shouldn't. The `functionObject`config should always override. And that's missing..

The right logic would be:
```
// You can do this before the loop
const includeByDefault = (this.warmup.includeAll === true ||
           (Array.isArray(this.warmup.includeAll) &&
             this.warmup.includeAll.indexOf(this.options.stage) !== -1)) 

//And the logic to include or not should be something like 
const excluded = functionObject.warmup !== true &&
          functionObject.warmup !== this.options.stage &&
          (!Array.isArray(functionObject.warmup) || 
            functionObject.warmup.indexOf(this.options.stage) === -1)) &&
          !includeByDefault

        if (!excluded) {
          return functionObject
        }
```

Or something like that...

Reply:
Good points. I'll take another look at this in the next few days and update it.

Reply:
My apologies for not getting back to this. Thanks again for doing this.